domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaygroup/TabShowConditionValidatorImpl.java

```diff
@@ -0,0 +1,103 @@
+package uk.gov.hmcts.ccd.definition.store.domain.validation.displaygroup;
+
+import com.google.common.collect.Lists;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.definition.store.domain.showcondition.InvalidShowConditionException;
+import uk.gov.hmcts.ccd.definition.store.domain.showcondition.ShowCondition;
+import uk.gov.hmcts.ccd.definition.store.domain.showcondition.ShowConditionParser;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationResult;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.DisplayGroupCaseFieldEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.DisplayGroupEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.DisplayGroupType;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+@Component
+public class TabShowConditionValidatorImpl implements DisplayGroupValidator {
+
+    private final ShowConditionParser showConditionParser;
+
+    @Autowired
+    public TabShowConditionValidatorImpl(ShowConditionParser showConditionParser) {
+        this.showConditionParser = showConditionParser;
+    }
+
+    @Override
+    public ValidationResult validate(DisplayGroupEntity thisDisplayGroup, Collection<DisplayGroupEntity> allDisplayGroups) {
```

Comment:
we we are using Collection? 
Collection interface is very limited, usually List is preferrable